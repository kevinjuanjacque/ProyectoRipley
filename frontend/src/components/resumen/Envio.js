import { AiFillHome } from "react-icons/ai";
import { FaStore } from "react-icons/fa";
export const Envio = ({shipping}) => {

    const { retiroTienda,envioDomicilio } = shipping;
    return (
        <>
        <div className="row box-envio">
            <div className="col">
                <AiFillHome className="mt-2" size="50" color={(envioDomicilio) ? "green" : "red"} />
            </div>
            <div className=" col-10 price-font" >
                Despacho a domicilio
                <p id="DomicilioText" className={(envioDomicilio) ? "envio-font" :  "envio-font-false "}>{(envioDomicilio) ? 'Disponible' : 'No disponible'}</p>
            </div>
            
        </div>
        <div className="row box-envio">
            <div className="col">
                <FaStore className="mt-2" size="50" color={(retiroTienda) ? "green" : "red"} />
            </div>
            <div className=" col-10 price-font" >
                Retiro en tienda
                <p id="TiendaText" className={(retiroTienda) ? "envio-font" :  "envio-font-false "}>{(retiroTienda) ? 'Disponible' : 'No disponible'}</p>
            </div>
            
        </div>
        </>
    )
}
