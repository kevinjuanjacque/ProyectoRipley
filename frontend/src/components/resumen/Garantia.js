

export const Garantia = ({garantia}) => {
    return (
        <div className="background-garantia description-font">
            
            <p id="desc">{garantia.shortDescription}</p>
            <div className="row">
                <div className="col">
                    <div className="form-group form-check">
                        <input type="checkbox" className="form-check-input" id="exampleCheck1"/>
                        <label className="form-check-label" htmlFor="exampleCheck1">Elegir Garantia</label>
                    </div>
                </div>
                <div className="col">
                    <p id="price">{garantia.prices.priceGarantia}</p>
                </div>  
            </div>
        </div>
    )
}
