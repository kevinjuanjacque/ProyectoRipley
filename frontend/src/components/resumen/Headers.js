

export const Headers = ({name,sku,shortDescription}) => {
    return (
        <div>
            <h1 className="title">{name}</h1>
            <small className="sku-font">SKU: { sku }</small>
            <p className="description-font">{shortDescription}</p>
        </div>
    )
}
