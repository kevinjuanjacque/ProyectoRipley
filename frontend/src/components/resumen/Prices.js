import { useEffect, useState } from "react"

export const Prices = ({prices}) => {
    const [Offert, setOffert] = useState();
    const [BetsPrice, setBetsPrice] = useState({Normal:prices.priceList});
    
    useEffect(() => {
        if(parseInt(prices.discountPorcentage)>0){
            setOffert(true);
            let Descuento = [prices.priceCard,prices.priceOffert].sort((a,b)=>a-b)[0];
            if(Descuento === prices.priceCard){
                setBetsPrice({card:Descuento});
            }else{
                setBetsPrice({internet:Descuento});
            }
        
        }else{
            setOffert(false);
        }
        
    }, [prices.discountPorcentage,prices.priceCard,prices.priceOffert])
    

    return (
        <div>

            <div className={(Offert) ? "row price-font" : "row best-price-font"}>

                <div className="col">
                    <span id="precio">{(Offert) ? 'Normal' : 'Internet' }</span>
                </div>
                <div className="col">
                    <span> { (!Offert) ? prices.formatted.priceList : <strike>{prices.formatted.priceList} </strike>} </span>
                </div>

            </div>

            {
                (Offert) && <>
                    <div className= "row best-price-font">

                        <div className="col">
                            <span >{(BetsPrice.card) ? 'Tarjeta Ripley' : 'Internet'}</span>
                        </div>
                        <div className="col">
                            <span >{ (BetsPrice.card) ? prices.formatted.priceCard : prices.formatted.priceOffer }</span>
                        </div>

                    </div>
                    <div className="row price-font">
                        <div className="col">
                            Descuento
                        </div>
                        <div className="col">
                            <p className="discount-font-porcentage">

                                { prices.discountPorcentage}%
                            </p>
                        </div>

                    </div>
                </>
            }

        </div>
    )
}
