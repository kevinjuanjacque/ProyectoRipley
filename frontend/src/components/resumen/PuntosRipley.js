
export const PuntosRipley = ({ripleyPuntos}) => {
    return (
        <div className="row description-font background-puntos">

            <div className="col ">
                Con tarjeta Ripley acumulas
            </div>
            <div className="col">
                <strong>{ripleyPuntos}</strong> puntos
            </div>
            
        </div>
    )
}
