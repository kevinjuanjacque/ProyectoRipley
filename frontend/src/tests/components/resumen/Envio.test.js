
import { Envio } from '../../../components/resumen/Envio';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

describe('Pruebas en <Envio.js />', () => {
    
    test('Debe renderizar <Envio.js /> correctamente', () => {
        
        const shipping = {
            retiroTienda:true,
            envioDomicilio:true
        } 


        const wrapper = shallow( <Envio shipping={shipping} />)
        expect( wrapper ).toMatchSnapshot();

    });

    test('Prueba del parrafo envio a domicilio ', () => {
        
        const shipping = {
            retiroTienda:true,
            envioDomicilio:true
        } 


        const wrapper = shallow( <Envio shipping={shipping} />)

        const parrafoDomicilio = wrapper.find("#DomicilioText").text();

        expect( parrafoDomicilio ).toBe((shipping.envioDomicilio) ? 'Disponible' : 'No disponible');

    })
    test('Prueba del parrafo retiro en tienda ', () => {

        const shipping = {
            retiroTienda:true,
            envioDomicilio:true
        } 


        const wrapper = shallow( <Envio shipping={shipping} />)

        const parrafoDomicilio = wrapper.find("#TiendaText").text();

        expect( parrafoDomicilio ).toBe((shipping.retiroTienda) ? 'Disponible' : 'No disponible');

    })
    
    
})
