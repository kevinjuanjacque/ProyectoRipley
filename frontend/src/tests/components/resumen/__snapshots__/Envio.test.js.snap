// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Envio.js /> Debe renderizar <Envio.js /> correctamente 1`] = `
<Fragment>
  <div
    className="row box-envio"
  >
    <div
      className="col"
    >
      <AiFillHome
        className="mt-2"
        color="green"
        size="50"
      />
    </div>
    <div
      className=" col-10 price-font"
    >
      Despacho a domicilio
      <p
        className="envio-font"
        id="DomicilioText"
      >
        Disponible
      </p>
    </div>
  </div>
  <div
    className="row box-envio"
  >
    <div
      className="col"
    >
      <FaStore
        className="mt-2"
        color="green"
        size="50"
      />
    </div>
    <div
      className=" col-10 price-font"
    >
      Retiro en tienda
      <p
        className="envio-font"
        id="TiendaText"
      >
        Disponible
      </p>
    </div>
  </div>
</Fragment>
`;
